appcues/src/test/java/com/appcues/statemachine/states/StatesTestingHelper.kt

```diff
@@ -0,0 +1,57 @@
+package com.appcues.statemachine.states
+
+import com.appcues.statemachine.Action
+import com.appcues.statemachine.Action.EndExperience
+import com.appcues.statemachine.Action.MoveToStep
+import com.appcues.statemachine.Action.RenderStep
+import com.appcues.statemachine.Action.ReportError
+import com.appcues.statemachine.Action.Reset
+import com.appcues.statemachine.Action.StartExperience
+import com.appcues.statemachine.Action.StartStep
+import com.appcues.statemachine.Error
+import com.appcues.statemachine.SideEffect
+import com.appcues.statemachine.State
+import com.appcues.statemachine.Transition
+import com.google.common.truth.Truth.assertThat
+import io.mockk.mockk
+
+internal fun State.assertIgnoredActions(list: List<Action>) {
+    list.forEach { assertThat(take(it)).isNull() }
+}
+
+internal fun Transition?.assertState(state: State) {
+    if (this != null) {
+        assertThat(newState).isEqualTo(state)
+    } else {
+        assert(false) { "transition should not be null" }
+    }
+}
+
+internal fun Transition?.assertEffect(effect: SideEffect?) {
+    if (this != null) {
+        assertThat(error).isNull()
+        assertThat(sideEffect).isEqualTo(effect)
+    } else {
+        assert(false) { "transition should not be null" }
+    }
+}
+
+internal fun Transition?.assertError(error: Error) {
```

Comment:
when asserting error, side effect should be null (it will be ignored anyways but just in case we are validating here)